File: src/abstract_factory/base/Maze.java
package abstract_factory.base;

import java.util.ArrayList;
import java.util.List;

public class Maze {
    private List<Room> roomList = new ArrayList<>();

    public void addRoom(Room room) {
        roomList.add(room);
    }

    public Room getRoom(int roomNumber) {
        return roomList.stream().filter(room -> room.getNumber() == roomNumber).findFirst().get();
    }
}

File: src/abstract_factory_kotlin/base/Maze.kt
package abstract_factory_kotlin.base

import kotlin.collections.ArrayList

class Maze {
    private val roomList = ArrayList<Room>()
    fun addRoom(room: Room) { roomList.add(room) }
    fun getRoom(roomNumber: Int) = roomList.find { room -> room.number == roomNumber }
}
